cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -296,6 +300,7 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.kafkaRoutes())
                 .compose(state -> state.kafkaIngresses())
                 .compose(state -> state.kafkaInternalServicesReady())
+                .compose(state -> state.kafkaGetClusterId())
```

Comment:
If we're going to change the exception handling, perhaps this should run last, or at least after `kafkaPodsReady`.